quickwit/quickwit-ui/src/components/QueryEditor/AggregationEditor.tsx

```diff
@@ -0,0 +1,316 @@
+      newAggregations[pos] = {histogram: {interval:value}};
+      updateAggregationProp(newAggregations);
+      return newAggregations;
+    });
+  }
+
+  const handleTermFieldChange = (pos: number, event: React.ChangeEvent<HTMLInputElement|HTMLTextAreaElement>) => {
+    const value = event.target.value;
+    setAggregations((agg) => {
+      const newAggregations = [...agg];
+      const term = newAggregations[pos]
+      if (isTerm(term)) {
+        term.term.field = value;
+      }
+      updateAggregationProp(newAggregations);
+      return newAggregations;
+    });
+  };
+
+  const handleTermCountChange = (pos: number, event: React.ChangeEvent<HTMLInputElement|HTMLTextAreaElement>) => {
+    const value = event.target.value;
+    setAggregations((agg) => {
+      const newAggregations = [...agg];
+      const term = newAggregations[pos]
+      if (isTerm(term)) {
+        term.term.size = Number(value);
+      }
+      updateAggregationProp(newAggregations);
+      return newAggregations;
+    });
+  };
+
+  function isHistogram(agg: {term: TermAgg} | {histogram: HistogramAgg} | undefined): agg is {histogram: HistogramAgg} {
+    if (!agg) return false;
+    return "histogram" in agg;
+  }
+
+  function isTerm(agg: {term: TermAgg} | {histogram: HistogramAgg} | undefined): agg is {term: TermAgg} {
+    if (!agg) return false;
+    return "term" in agg;
+  }
+
+  const getAggregationKind = (agg: {term: TermAgg} | {histogram: HistogramAgg} | undefined) => {
+    if (isHistogram(agg)) {
+      return "histogram";
+    }
+    if (isTerm(agg)) {
+      return "term";
+    }
+    return "new";
+  };
+
+  const makeOptions = (pos: number, agg: ({term: TermAgg} | {histogram: HistogramAgg})[]) => {
+    const options = [];
+    if (pos >= agg.length) {
+      options.push((
+          <MenuItem value="new" key="new">Add aggregation</MenuItem>
+      ))
+    }
+    let addHistogram = true;
+    let addTerm = true;
+    for(let i = 0; i < agg.length; i++) {
+      if (i == pos) continue;
+      if (getAggregationKind(agg[i]) === "histogram") addHistogram = false;
+      if (getAggregationKind(agg[i]) === "term") addTerm = false;
+    }
+    if (addHistogram) {
+      options.push((<MenuItem value="histogram" key="histogram">Histogram aggregation</MenuItem>))
+    }
+    if (addTerm) {
+      options.push((<MenuItem value="term" key="term">Term aggregation</MenuItem>));
+    }
+    if (agg.length > 1) {
+      options.push((
+          <MenuItem value="rm" key="rm">Remove aggregation</MenuItem>
+      ))
+    }
+    return options;
+  }
+
+  // do the initial filling of parameters
+  const aggregationConfig = props.searchRequest.aggregationConfig;
+  if (aggregationConfig.histogram === null && aggregationConfig.term === null) {
+    const initialAggregation = Object.assign({}, ...aggregations);
+    const initialSearchRequest = {...props.searchRequest, aggregationConfig: initialAggregation};
+    props.onSearchRequestUpdate(initialSearchRequest);
```

Comment:
This will cause a setState in the parent component from the render fn of the current one, which triggers a render somewhere else in the tree. Undesired side-effect inside a render, not allowed (react will warn in the console).

You can wrap this init block in a `useEffect(()=>{/*init here*/},[/*deps*/])` call. Leave the deps array empty if the effect is to be executed once.